runtime/include/tasks/qthreads/chpl-tasks-impl-fns.h

```diff
@@ -173,8 +173,6 @@ void chpl_task_setSubloc(c_sublocid_t full_subloc)
 {
     qthread_shepherd_id_t curr_shep;
 
-    assert(isActualSublocID(full_subloc) || full_subloc == c_sublocid_any);
-
```

Comment:
Unless I missed something, you aren't modifying anything in the NUMA locale model. Would removing this assertion cause any issues for it?

Reply:
This assertion gets triggered for the GPU locale model because we end up using `c_sublocid_none` to represent the CPU.

Maybe in the NUMA locale model this would be considered an invalid state. I suppose I could update the assertion to only apply if `CHPL_LOCALE_MODEL=numa` but I'm not sure how much value there would be in doing that.

Reply:
I am not sure either. I'd still be inclined to do that to make sure that we are not leaving NUMA in a worse state than it is already.

Reply:
Ok, I changed the assertion to this:

```C++
    // We allow using c_sublocid_none to represent the CPU in the gpu locale
    // model. This isn't currently used by the numa (or other locale) models.
    assert(isActualSublocID(full_subloc) || full_subloc == c_sublocid_any ||
        !strcmp(CHPL_LOCALE_MODEL, "gpu"));
```